### PR TITLE
(APS-462/APS-464) Add mental health and recovery-focussed AP types

### PR DESCRIPTION
**Pull request by yndajas**
# Context

[JIRA](https://dsdmoj.atlassian.net/browse/APS-462)

Closes #1595, closes #1632, closes #1650

# Changes in this PR

- Switch to AP type enum/union in Apply
- Add mental health and recovery-focussed AP types to screens:
  - Apply type of AP
  - Assess matching information
- Small refactors

#### Before

image

#### After

image

image

image

image

#### Before

image

#### After

image

### CODE REVIEW

File: server/utils/applications/utils.ts [hunk 1]
```diff
@@ -258,10 +258,27 @@ const firstPageOfApplicationJourney = (application: Application) => {
...
+      return 'MHAP (Elliott House)'
+    case 'mhapStJosephs':
+      return 'MHAP (St Josephs)'
+    case 'pipe':
+      return 'PIPE'
+    case 'rfap':
+      return 'RFAP'
+    default:
+      return 'Standard'
+  }
```
Comment (yndajas) [inline server/utils/applications/utils.ts:281]: CONFIRM: is this expected behaviour? The function is currently used for allocation and task summaries

File: integration_tests/fixtures/applicationData.json [hunk 2]
```diff
@@ -90,7 +90,7 @@
   },
   "type-of-ap": {
     "ap-type": {
-      "type": "standard"
```
Comment (yndajas) [inline integration_tests/fixtures/applicationData.json:93]: Soz for the big commit: changing things on the Apply page has a lot of repercussions!

File: server/utils/assessments/utils.ts [hunk 3]
```diff
@@ -41,10 +41,6 @@ const groupAssessmements = (assessments: Array<AssessmentSummary>): GroupedAsses
   return result
 }
 
-const getApplicationType = (assessment: Assessment): ApplicationType => {
-  return getApplicationTypeFromApplication(assessment.application)
```
Comment (edavey) [inline server/utils/assessments/utils.ts:45]: thanks for simplifying

File: server/utils/applications/utils.ts (hunk 1 above)
Comment (edavey) [reply server/utils/applications/utils.ts:281]: yes, this looks correct to me

File: server/utils/applications/getApplicationData.test.ts [hunk 4]
```diff
@@ -59,162 +69,58 @@ describe('getApplicationData', () => {
...
-    it('returns in_community for a bail placement application', () => {
-      mockQuestionResponse({
-        sentenceType: 'bailPlacement',
-        postcodeArea: targetLocation,
-        situation: 'riskManagement',
-        apAreaId,
-        applicantUserDetails,
-        caseManagerIsNotApplicant: false,
-      })
+    it('returns the correct data for a bail placement application', () => {
```
Comment (edavey) [inline server/utils/applications/getApplicationData.test.ts:96]: thanks for improving these descriptions

**Review by edavey — APPROVED** [review]
This looks good to me. Thanks for the various refactorings and improvements.  

As this isn't behind a feature flag we'll need to be clear that it will be included in the next release.

**Pull request merged**